lib/hocon/impl/simple_config_object.rb

```diff
@@ -181,6 +187,62 @@ def key_set
     Set.new(@value.keys)
   end
 
+  def self.map_hash(m)
+    # the keys have to be sorted, otherwise we could be equal
+    # to another map but have a different hashcode.
+    keys = m.keys.sort
+
+    value_hash = 0
+
+    keys.each do |key|
+      value_hash += m[key].hash
+    end
+
+    41 * (41 + keys.hash) + value_hash
+  end
+
+  def self.map_equals(a, b)
+    # This array comparison works if there are no duplicates, which
+    # the hash keys won't have
+    sets_equal = lambda { |x, y| (x.size == y.size) && (x & y == x) }
+
+    if a == b
+      return true
+    end
+
+    if not sets_equal.call(a.keys, b.keys)
```

Comment:
`a.keys == b.keys` will be false if the order of the elements is different, and Hash#keys doesn't guarantee an order. I think the java code compares two sets instead of two arrays, so the order doesn't matter in the comparison